community/kernel/src/main/java/org/neo4j/kernel/impl/traversal/TraversalBranchImpl.java

```diff
@@ -174,7 +157,24 @@ protected TraversalBranch newNextBranch( Node node, Relationship relationship )
     @Override
     public void prune()
     {
-        relationships = PRUNED_ITERATOR;
+        closeRelationships();
+        relationships = Iterators.emptyResourceIterator();
+    }
+
+    @Override
+    public void close()
+    {
+        //TODO: not sure its a right place to close parent
+        parent.close();
```

Comment:
It feels iffy to close parent here because just because this branch is exhausted doesn't mean that parent it.